test/sw_mgmt_spec.rb

```diff
@@ -367,4 +367,69 @@
       subject.zypp_config_writable!
     end
   end
+
+  describe ".check_repositories" do
+    let(:repo) { 42 }
+
+    before do
+      allow(Yast::Pkg).to receive(:SourceGetCurrent).with(true).and_return([repo])
+      allow(Yast::Pkg).to receive(:SourceGeneralData).with(repo)
+        .and_return("autorefresh" => true)
+    end
+
+    it "refreshes all repositories with autorefresh enabled" do
+      expect(Yast::Pkg).to receive(:SourceGeneralData).with(repo)
+        .and_return("autorefresh" => true)
+      expect(Yast::Pkg).to receive(:SourceRefreshNow).with(repo).and_return(true)
+
+      subject.check_repositories
+    end
+
+    it "returns true if all repositores refresh" do
+      expect(Yast::Pkg).to receive(:SourceRefreshNow).with(repo).and_return(true)
+
+      expect(subject.check_repositories).to eq(true)
+    end
+
+    context "a repository refresh fails" do
+      before do
+        expect(Yast::Pkg).to receive(:SourceRefreshNow).with(repo).and_return(false)
+        allow(Yast::Popup).to receive(:ErrorAnyQuestion).and_return(false)
```

Comment:
Good point, I wanted to share this code, but in the end I used a different mock in each test and this line become useless. I'll remove it.